lib/active_record_shards/connection_switcher.rb

```diff
@@ -95,21 +95,15 @@ def on_master(&block)
     alias_method :with_slave_if, :on_slave_if
     alias_method :with_slave_unless, :on_slave_unless
 
-    def on_cx_switch_block(which, options = {}, &block)
+    def on_cx_switch_block(which, _options = {}, &_block)
```

Comment:
I think we can remove the explicit block if we don’t use it.

And since this is a breaking change anyway, how about removing the `_options` as well, and change the few internal callers of this method?